orangecontrib/infrared/widgets/owfft.py

```diff
@@ -0,0 +1,594 @@
+import sys
+import numpy as np
+
+from PyQt4.QtGui import QGridLayout
+
+import Orange.data
+from Orange.widgets.widget import OWWidget
+from Orange.widgets import gui, settings
+
+def build_spec_table(wavenumbers, intensities):
+    """
+    Converts numpy arrays of wavenumber and intensity into an
+    Orange.data.Table spectra object.
+
+    Args:
+        wavenumbers (np.array): 1D array of wavenumbers
+        intensities (np.array): 2D array of (multi-spectra) intensities
+
+    Returns:
+        table: Orange.data.Table object, spectra format
+    """
+
+    # Add dimension to 1D array if necessary
+    if intensities.ndim == 1:
+        intensities = intensities[None,:]
+
+    # Convert the wavenumbers array into a list of ContinousVariables
+    wn_vars = [Orange.data.ContinuousVariable.make(repr(wavenumbers[i]))
+                for i in range(wavenumbers.shape[0])]
+
+    # Build an Orange.data.Domain object with wn_vars as
+    # independant variables (or "attributes" as Orange calls them)
+    domain = Orange.data.Domain(wn_vars)
+
+    # Finally, build the table using the damain and intensity arrays:
+    table = Orange.data.Table.from_numpy(domain, intensities)
+    return table
+
+class OWFFT(OWWidget):
+    # Widget's name as displayed in the canvas
+    name = "FFT"
+
+    # Short widget description
+    description = (
+        "Performs Fast Fourier Transform on an interferogram, including "
+        "zero filling, apodization and phase correction.")
+
+    # An icon resource file path for this widget
+    # (a path relative to the module where this widget is defined)
+    icon = "icons/fft.svg"
+
+    # Define inputs and outputs
+    inputs = [("Interferogram", Orange.data.Table, "set_data")]
+    outputs = [("Spectra", Orange.data.Table)]
+
+    # Define widget settings
+    laser_wavenumber = settings.Setting(15797.337544)
```

Comment:
​They measure the point spacing with an optical encoder usually, so they
just give you a spacing number in nm (or meter)​


On Tue, May 31, 2016 at 4:41 PM, Stuart Read <notifications@github.com>
wrote:

> In orangecontrib/infrared/widgets/owfft.py
> <https://github.com/markotoplak/orange-infrared/pull/8#discussion_r65194621>
> :
>
> > +
> > +    # Short widget description
> > +    description = (
> > +        "Performs Fast Fourier Transform on an interferogram, including "
> > +        "zero filling, apodization and phase correction.")
> > +
> > +    # An icon resource file path for this widget
> > +    # (a path relative to the module where this widget is defined)
> > +    icon = "icons/fft.svg"
> > +
> > +    # Define inputs and outputs
> > +    inputs = [("Interferogram", Orange.data.Table, "set_data")]
> > +    outputs = [("Spectra", Orange.data.Table)]
> > +
> > +    # Define widget settings
> > +    laser_wavenumber = settings.Setting(15797.337544)
>
> Interesting. I could change this setting to be directly the dx data point
> spacing, however that would be awkward for users with lasers specified in cm
> -1. How is the spacing defined in the NanoIR instrument?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/markotoplak/orange-infrared/pull/8/files/1cc2c433b5fee6b038065bfdc91998f0f6a8e1b6#r65194621>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGA0GtDvhjrDV6K4kf_xZ36VTJ0PISdAks5qHEiUgaJpZM4IqDpJ>
> .
>